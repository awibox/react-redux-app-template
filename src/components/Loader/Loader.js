import React from 'react';
// Styles
import styles from './Loader.scss';

const Loader = () => (
  <div className={styles.loader}>
    <div className={styles.spinnerBorder} />
  </div>
);

export default Loader;
