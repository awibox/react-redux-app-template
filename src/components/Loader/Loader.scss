@import '../../styles/variables';

$spinnerSize: 50px;

.loader {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 99;
  background: rgba(255,255,255,.5);
}

%clearFix {
  content: "";
  position: relative;
  display: block;
}

.spinnerBorder {
  position: absolute;
  left:50%;
  top: 50%;
  transform: translate(-50%, -50%);
  //width: $spinnerSize;
  //height: $spinnerSize;
  &:before {
    @extend %clearFix;
    animation: spinner 2.5s cubic-bezier(0.75, 0, 0.5, 1) infinite normal;
    width: $spinnerSize;
    height: $spinnerSize;
    background-color: $navActiveBackground;
  }
  &:after {
    @extend %clearFix;
    animation: shadow 2.5s cubic-bezier(0.75, 0, 0.5, 1) infinite normal;
    bottom: -$spinnerSize/2;
    height: $spinnerSize/4;
    border-radius: 50%;
    background-color: rgba($navActiveBackground,0.2);
  }
}

@keyframes spinner {
  50% {
    border-radius: 50%;
    transform: scale(0.5) rotate(360deg);
  }
  100% {
    transform: scale(1) rotate(720deg);
  }
}
@keyframes shadow {
  50% {
    transform: scale(0.5);
    background-color: rgba(#000,0.1);
  }
}
