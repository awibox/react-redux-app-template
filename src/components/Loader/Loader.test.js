import React from 'react';
import { shallow } from 'enzyme';
import Loader from './Loader';

describe('Loader', () => {
  it('should render correctly', () => {
    const component = shallow(<Loader/>);
    expect(component).toMatchSnapshot();
  });
});
