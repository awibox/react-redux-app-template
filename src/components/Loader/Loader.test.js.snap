// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render correctly 1`] = `
<div
  className="loader"
>
  <div
    className="spinnerBorder"
  />
</div>
`;
